### PR TITLE
chore(lambda): add toString to lambda architecture

### LINKED ISSUE #26117: lambda: Updating The Lambda Architecture Properties with toString() methods
### Describe the feature

Updating The Lambda Properties with toString() methods

### Use Case

We want to use effective error throwing using config properties, but they require a lot of code that could be decluttered if the properties have a toString method consistently (like there is on Runtime)

### Proposed Solution

Add toString() methods to the Lambda Props

### Other Information

_No response_

### Acknowledgements

- [X] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.69.0

### Environment details (OS name and version, etc.)

macOS Ventura -  Version 13.3.1 (22E261)

### PR DESCRIPTION
**Pull request by lukey-aleios**
Add  toString to Lambda Architecture property

When throwing errors during synthesis, you can't see the architecture chosen for a lambda. Adding this toString method solves that

Closes #26117

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/aws-cdk-lib/aws-lambda/lib/architecture.ts [hunk 1]
```diff
@@ -36,4 +36,8 @@ export class Architecture {
     this.name = archName;
     this.dockerPlatform = dockerPlatform;
   }
+
+  public toString(): string {
```
Comment (corymhall) [inline packages/aws-cdk-lib/aws-lambda/lib/architecture.ts:40]: Can you add a docstring here?

**Review by corymhall — CHANGES_REQUESTED** [review]
Can you add a unit test for this?

File: packages/aws-cdk-lib/aws-lambda/lib/architecture.ts (hunk 1 above)
Comment (lukey-aleios) [reply packages/aws-cdk-lib/aws-lambda/lib/architecture.ts:40]: Yep! Great point

**Review by mrgrain — DISMISSED** [review]
@lukey-aleios How are you doing with the unit test? If you need any help checkout the Community Slack on cdk.dev channel #contributing

**Comment by lukey-aleios** [comment]
> @lukey-aleios How are you doing with the unit test? If you need any help checkout the Community Slack on cdk.dev channel #contributing

Hey! I've been away for the past two weeks so will be able to get to this early next week 😄

**Comment by mrgrain** [comment]
> Hey! I've been away for the past two weeks so will be able to get to this early next week 😄

Awesome, thank you!

**Comment by mrgrain** [comment]
If the automation closes your PR, please just open a new one once you have had time to address any issues. Pro tip: Open the PR in Draft mode and our automation won't touch it.

**Comment by mrgrain** [comment]
Your test case doesn't quite work out
```
aws-cdk-lib: FAIL aws-lambda/test/architecture.test.js
aws-cdk-lib:   â— architecture â€º toString to return the architecture name
aws-cdk-lib:     expect(received).toEqual(expected) // deep equality
aws-cdk-lib:     Expected: "x86_64"
aws-cdk-lib:     Received: {"dockerPlatform": "linux/amd64", "name": "x86_64"}
aws-cdk-lib:     [0m [90m 17 |[39m[0m
aws-cdk-lib:     [0m [90m 18 |[39m     [90m// THEN[39m[0m
aws-cdk-lib:     [0m[31m[1m>[22m[39m[90m 19 |[39m     expect(testLambda[33m.[39marchitecture)[33m.[39mtoEqual(testLambda[33m.[39marchitecture[33m.[39mname)[33m;[39m[0m
aws-cdk-lib:     [0m [90m    |[39m                                     [31m[1m^[22m[39m[0m
aws-cdk-lib:     [0m [90m 20 |[39m   })[33m;[39m[0m
aws-cdk-lib:     [0m [90m 21 |[39m })[33m;[39m[0m
aws-cdk-lib:     [0m [90m 22 |[39m[0m
```

**Comment by lukey-aleios** [comment]
> Your test case doesn't quite work out
> 
> ```
> aws-cdk-lib: FAIL aws-lambda/test/architecture.test.js
> aws-cdk-lib:   â—� architecture â€º toString to return the architecture name
> aws-cdk-lib:     expect(received).toEqual(expected) // deep equality
> aws-cdk-lib:     Expected: "x86_64"
> aws-cdk-lib:     Received: {"dockerPlatform": "linux/amd64", "name": "x86_64"}
> aws-cdk-lib:     �[0m �[90m 17 |�[39m�[0m
> aws-cdk-lib:     �[0m �[90m 18 |�[39m     �[90m// THEN�[39m�[0m
> aws-cdk-lib:     �[0m�[31m�[1m>�[22m�[39m�[90m 19 |�[39m     expect(testLambda�[33m.�[39marchitecture)�[33m.�[39mtoEqual(testLambda�[33m.�[39marchitecture�[33m.�[39mname)�[33m;�[39m�[0m
> aws-cdk-lib:     �[0m �[90m    |�[39m                                     �[31m�[1m^�[22m�[39m�[0m
> aws-cdk-lib:     �[0m �[90m 20 |�[39m   })�[33m;�[39m�[0m
> aws-cdk-lib:     �[0m �[90m 21 |�[39m })�[33m;�[39m�[0m
> aws-cdk-lib:     �[0m �[90m 22 |�[39m�[0m
> ```

Should be fixed now!

**Review by mrgrain — APPROVED** [review]

**Pull request merged**